homeassistant/components/ezviz/camera.py

```diff
@@ -0,0 +1,276 @@
+                "localRtspPort": local_rtsp_port,
+                "netType": connections[device_serial]["netType"],
+                "wanIp": connections[device_serial]["wanIp"],
+            }
+
+        # Add the cameras as devices in HASS
+        for camera_serial in cameras:
+            camera = cameras[camera_serial]
+            _LOGGER.debug("CAMERA: %s", camera)
+
+            camera_username = CONF_DEFAULT_CAMERA_USERNAME
+            camera_password = ""
+            camera_rtsp_stream = ""
+
+            if camera_serial in conf_cameras:
+                camera_username = conf_cameras[camera_serial]["username"]
+                camera_password = conf_cameras[camera_serial]["password"]
+                camera_rtsp_stream = "rtsp://{}:{}@{}:{}".format(
+                    camera_username,
+                    camera_password,
+                    camera["localIp"],
+                    camera["localRtspPort"],
+                )
+                _LOGGER.debug(
+                    "Camera %s source stream: %s", camera["serial"], camera_rtsp_stream
+                )
+
+            else:
+                _LOGGER.error(
+                    "I found a camera (%s) but it is not configured. Please configure it if you wish to see the appropriate stream. Conf cameras: %s",
+                    camera_serial,
+                    conf_cameras,
+                )
+
+            async_add_entities(
+                [
+                    EzvizCamera(
+                        hass,
+                        camera_username,
+                        camera_password,
+                        camera_rtsp_stream,
+                        camera["serial"],
+                        camera["name"],
+                        camera["deviceType"],
+                        camera["version"],
+                        camera["status"],
+                        camera["createTime"],
+                        camera["category"],
+                        camera["subCategory"],
+                        camera["customType"],
+                        camera["localIp"],
+                        camera["netIp"],
+                        camera["localRtspPort"],
+                        camera["wanIp"],
+                        camera["netType"],
+                    )
+                ]
+            )
+    except PyEzvizError as exp:
+        _LOGGER.error(exp)
+        return
+
+
+class EzvizCamera(Camera):
+    """An implementation of a Foscam IP camera."""
+
+    def __init__(
+        self,
+        hass,
+        username,
+        password,
+        rtsp_stream,
+        serial,
+        name,
+        device_type,
+        version,
+        status,
+        create_time,
+        category,
+        sub_category,
+        custom_type,
+        local_ip,
+        net_ip,
+        local_rtsp_port,
+        wan_ip,
+        net_type,
+    ):
+        """Initialize an Ezviz camera."""
+        super().__init__()
```

Comment:
Remove, this is automatically called.

Reply:
We have to call it explicitly since the parent has an init method.

Reply:
Ok, so I leave it, right ?

Reply:
Yes.